depthai_helpers/arg_manager.py

```diff
@@ -0,0 +1,189 @@
+import os
+import argparse
+try:
+    import argcomplete
+except ImportError:
+    raise ImportError('argcomplete module not found, run python3 -m pip install -r requirements.txt ')
+from argcomplete.completers import ChoicesCompleter
+
+from depthai_helpers.cli_utils import cli_print, PrintColors
+import consts.resource_paths
+
+
+def _get_immediate_subdirectories(a_dir):
+    return [name for name in os.listdir(a_dir)
+            if os.path.isdir(os.path.join(a_dir, name))]
+
+_stream_choices = ("metaout", "previewout", "jpegout", "left", "right", "depth_raw", "disparity", "disparity_color", "meta_d2h", "object_tracker")
+_CNN_choices = _get_immediate_subdirectories(consts.resource_paths.nn_resource_path)
+
+def _stream_type(option):
+    max_fps = None
+    option_list = option.split(",")
+    option_args = len(option_list)
+    if option_args not in [1, 2]:
+        msg_string = "{0} format is invalid. See --help".format(option)
+        cli_print(msg_string, PrintColors.WARNING)
+        raise ValueError(msg_string)
+
+    deprecated_choices = ("depth_sipp", "depth_color_h")
+    transition_map = {"depth_sipp" : "disparity_color", "depth_color_h" : "disparity_color"}
+    stream_name = option_list[0]
+    if stream_name in deprecated_choices:
```

Comment:
Can be simplified
```suggestion
    if stream_name in transition_map:
```

Reply:
I'm not sure if I understand this simplification. It seems like switching deprecated_choices and transition_map in this conditional would cause totally different behavior.

Reply:
So in this case we want to check if one of the depreciated streams was passed by the user. We have transition map with keys being the old stream names and values being the new stream names

When we do `if key in dict` we check if the key is present in the dict's keys. So in this case, as you have listed all depreciated stream names as dict keys, you can go ahead and do `if stream in transition_map`.

`depreciated_choices`  is the same as `transition_map.keys()`, so `if stream_name in depreciated_choices` is equal to `if stream_name in transition_map.keys()`.
And since comparison `if stream_name in transition_map.keys()` is the same as `if stream_name in transition_map`, we can remove the `depreciated_choices` array and stick with just `transition_map`

Reply:
Oh, gotcha, didn't notice they contained the same keys. Yeah, I'll get rid deprecated_choices.